src/com/owncloud/android/files/services/FileUploadService.java

```diff
@@ -0,0 +1,1359 @@
+
+    private IBinder mBinder;
+    private OwnCloudClient mUploadClient = null;
+    private Account mLastAccount = null;
+    private FileDataStorageManager mStorageManager;
+    //since there can be only one instance of an Android service, there also just one db connection.
+    private UploadDbHandler mDb = null;
+    
+    private final AtomicBoolean mCancellationRequested = new AtomicBoolean(false);
+    private final AtomicBoolean mCancellationPossible = new AtomicBoolean(false);
+
+    /**
+     * List of uploads that are currently pending. Maps from remotePath to where file
+     * is being uploaded to {@link UploadFileOperation}.
+     */
+    private ConcurrentMap<String, UploadDbObject> mPendingUploads = new ConcurrentHashMap<String, UploadDbObject>();
+    
+    /**
+     * {@link UploadFileOperation} object of ongoing upload. Can be null. Note: There can only be one concurrent upload!
+     */
+    private UploadFileOperation mCurrentUpload = null;
+    
+    private NotificationManager mNotificationManager;
+    private NotificationCompat.Builder mNotificationBuilder;
+    private int mLastPercent;
+
+    private static final String MIME_TYPE_PDF = "application/pdf";
+    private static final String FILE_EXTENSION_PDF = ".pdf";
+
+    public static String getUploadFinishMessage() {
+        return FileUploadService.class.getName().toString() + UPLOAD_FINISH_MESSAGE;
+    }
+
+    /**
+     * Builds a key for mPendingUploads from the account and file to upload
+     * 
+     * @param account Account where the file to upload is stored
+     * @param file File to upload
+     */
+    private String buildRemoteName(Account account, OCFile file) {
+        return account.name + file.getRemotePath();
+    }
+
+    
+    private String buildRemoteName(Account account, String remotePath) {
+        return account.name + remotePath;
+    }
+
+    private String buildRemoteName(UploadDbObject uploadDbObject) {
+        return uploadDbObject.getAccountName() + uploadDbObject.getRemotePath();
+    }
+
+    /**
+     * Checks if an ownCloud server version should support chunked uploads.
+     * 
+     * @param version OwnCloud version instance corresponding to an ownCloud
+     *            server.
+     * @return 'True' if the ownCloud server with version supports chunked
+     *         uploads.
+     */
+    private static boolean chunkedUploadIsSupported(OwnCloudVersion version) {
+        return (version != null && version.compareTo(OwnCloudVersion.owncloud_v4_5) >= 0);
+    }
+
+    /**
+     * Service initialization
+     */
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        Log_OC.d(TAG, "mPendingUploads size:" + mPendingUploads.size() + " - onCreate");
+        mNotificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
+        mBinder = new FileUploaderBinder();
+        mDb = UploadDbHandler.getInstance(this.getBaseContext());
```

Comment:
Honestly, I have not really grasped the whole idea of Android's ´Context´. I just wanted to make sure, that `UploadDbHandler` always has the *same* context. (I thought it could store uploads per context otherwise which is not wanted.)